wfe/web-front-end_test.go

```diff
@@ -464,11 +465,29 @@ func TestIndex(t *testing.T) {
 	test.AssertEquals(t, responseWriter.Body.String(), "404 page not found\n")
 }
 
+func TestDirectory(t *testing.T) {
+	wfe := setupWFE(t)
+	wfe.BaseURL = "http://localhost:4300"
+	mux, err := wfe.Handler()
+	test.AssertNotError(t, err, "Failed to marshal directory to JSON")
+
+	responseWriter := httptest.NewRecorder()
+
+	url, _ := url.Parse("/directory")
+	mux.ServeHTTP(responseWriter, &http.Request{
+		Method: "GET",
+		URL:    url,
+	})
+	test.AssertEquals(t, responseWriter.Code, http.StatusOK)
+	test.AssertEquals(t, responseWriter.Body.String(), `{"new-authz":"http://localhost:4300/acme/new-authz","new-cert":"http://localhost:4300/acme/new-cert","new-reg":"http://localhost:4300/acme/new-reg","revoke-cert":"http://localhost:4300/acme/revoke-cert"}`)
+}
+
 // TODO: Write additional test cases for:
 //  - RA returns with a failure
 func TestIssueCertificate(t *testing.T) {
 	wfe := setupWFE(t)
-	mux := wfe.Handler()
+	mux, err := wfe.Handler()
+	test.AssertNotError(t, err, "Failed to marshal directory to JSON")
```

Comment:
Same issue with these errors - they should indicate failure of Handler(), rather than a more specific error.